pkg/ipam-node/controllers/cidrpool/cidrpool.go

```diff
@@ -0,0 +1,108 @@
+/*
+ Copyright 2023, NVIDIA CORPORATION & AFFILIATES
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"net"
+
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	ipamv1alpha1 "github.com/Mellanox/nvidia-k8s-ipam/api/v1alpha1"
+	"github.com/Mellanox/nvidia-k8s-ipam/pkg/common"
+	"github.com/Mellanox/nvidia-k8s-ipam/pkg/ip"
+	"github.com/Mellanox/nvidia-k8s-ipam/pkg/pool"
+)
+
+// CIDRPoolReconciler reconciles CIDRPool objects
+type CIDRPoolReconciler struct {
+	PoolManager pool.Manager
+	client.Client
+	Scheme   *runtime.Scheme
+	NodeName string
+}
+
+// Reconcile contains logic to sync CIDRPool objects
+func (r *CIDRPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	reqLog := log.FromContext(ctx)
+	cidrPool := &ipamv1alpha1.CIDRPool{}
+	poolKey := common.GetPoolKey(req.Name, common.PoolTypeCIDRPool)
+	err := r.Client.Get(ctx, req.NamespacedName, cidrPool)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			reqLog.Info("CIDRPool not found, removing from PoolManager")
+			r.PoolManager.RemovePool(poolKey)
+			return ctrl.Result{}, nil
+		}
+		reqLog.Error(err, "failed to get CIDRPool object from the cache")
+		return ctrl.Result{}, err
+	}
+	reqLog.Info("Notification on CIDRPool", "name", cidrPool.Name)
+	found := false
```

Comment:
nit: move found after the validation function to be closer to where it's used.

Reply:
done